spanner/google/cloud/spanner_v1/keyset.py

```diff
@@ -38,14 +38,19 @@ class KeyRange(object):
     """
     def __init__(self, start_open=None, start_closed=None,
                  end_open=None, end_closed=None):
-        if not any([start_open, start_closed, end_open, end_closed]):
+        keys = (start_open, start_closed, end_open, end_closed)
+        if all(key is None for key in keys):
```

Comment:
The presence / necessity of this check indicates the inherent "problem" with the approach of converting `start_open=None, start_closed=None` to an implicit `start_closed=[]` (similar for `end_*`).

Seeing this / realizing that only **one** of `start_` and `end_` can fall back to an empty value, it seems that the lesser of two evils is requiring the **user** to provide the `[]`.